### PR TITLE
Add some initial documentation for query caches

### PR DESCRIPTION
**Pull request by hedengran**

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-docs-cypher-521.surge.sh

**Comment by hedengran** [comment]
Questions to reviewers: Is this the correct location for the pages? I feel like putting it at the same level as "Syntax", "Clauses", "Administration", etc. might be shedding a bit too much light on a pretty advanced topic. Maybe it should even end up in the Operations Manual instead? It's the _"Cypher query caches"_, but it's also configurations at a DBMS level.

Note also that this is a first step in a bigger effort to thoroughly document query caches.

File: modules/ROOT/pages/query-caches/index.adoc [hunk 1]
```diff
@@ -0,0 +1,48 @@
+= Query caches
+
+Cypher utilizes a number of different caches to speed up query planning.
+
+Out of the box, the set of query caches are _per database_.
+That means that for each new database, a new set of caches will be initialized.
+
+The max number of entries per cache is configured with `server.memory.query_cache.per_db_cache_num_entries`.
```
Comment (sherfert) [inline modules/ROOT/pages/query-caches/index.adoc]:
```suggestion
The maximum number of entries per cache is configured with `server.memory.query_cache.per_db_cache_num_entries`.
```

File: modules/ROOT/pages/query-caches/unified-query-caches.adoc [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+| label:enterprise-edition[Enterprise only] Enable sharing cache space between different databases. With this option turned on, databases will share cache space,
+but not cache entries.
+
+| `server.memory.query_cache.shared_cache_num_entries`
+| `1000`
+| label:enterprise-edition[Enterprise only] The number of cached queries for all databases. This setting is only deciding cache size when
+`server.memory.query_cache.sharing_enabled` is set to `true`.
+|===
+
+When this feature is enabled, all databases will utilize only on set of query caches.
```
Comment (sherfert) [inline modules/ROOT/pages/query-caches/unified-query-caches.adoc]:
```suggestion
When this feature is enabled, all databases use only one set of query caches.
```

**Review by sherfert — CHANGES_REQUESTED** [review]
I think the location of the pages is good.

However, it seems the options are not described in the operations manual (https://neo4j.com/docs/operations-manual/current/reference/configuration-settings/) - at least not in the currently published version.

Is that already changed in `dev` of the operations manual? If not, I think this should go hand in hand. And then, from here, you can insert links to the relevant part of the website linked above.

**Comment by hedengran** [comment]
> However, it seems the options are not described in the operations manual

Yes, currently they are not. That's what I meant during standup this morning, that I just learned that they're not automatically added to where you linked. But @renetapopova is on it and will added soon I think. 

>  I think this should go hand in hand. And then, from here, you can insert links to the relevant part of the website linked above.

Yes that's a very good idea.

File: modules/ROOT/pages/query-caches/unified-query-caches.adoc [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+A database may store and retrieve entries from the shared cache, but it may not retrieve entries produced by another database.
+
+[IMPORTANT]
+====
+While databases utilize the same set of caches, a database may not observe entries originating from other databases.
```
Comment (renetapopova) [inline modules/ROOT/pages/query-caches/unified-query-caches.adoc]:
```suggestion
While databases use the same set of caches, a database may not observe entries originating from other databases.
```

File: modules/ROOT/pages/query-caches/unified-query-caches.adoc [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+====
+
+The database may, however, evict entries from other databases as necessary, according to the constrained cache size and cache eviction policy.
+In essence, databases may compete for cache space, but may not observe each other's entries.
+
+When this option is turned on, the cache space available to all databases is configured with
```
Comment (renetapopova) [inline modules/ROOT/pages/query-caches/unified-query-caches.adoc]:
```suggestion
When this option is turned on, the cache space available to all databases is configured with `server.memory.query_cache.shared_cache_num_entries`.
```

File: modules/ROOT/pages/query-caches/unified-query-caches.adoc [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+`server.memory.query_cache.shared_cache_num_entries`.
```
Comment (renetapopova) [inline modules/ROOT/pages/query-caches/unified-query-caches.adoc]:
```suggestion
```

File: modules/ROOT/pages/query-caches/index.adoc [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+[NOTE]
+====
+This setting is only deciding cache size when `server.memory.query_cache.sharing_enabled` is set to `false`.
+====
+
+Query caches may consume a lot of memory, especially when running many active databases.
+To combat this, and improve predictability on memory consumption,
```
Comment (renetapopova) [inline modules/ROOT/pages/query-caches/index.adoc]:
```suggestion
To combat this and improve the predictability of memory consumption, it is possible to configure the DBMS to use only one set of caches for all databases.
```

File: modules/ROOT/pages/query-caches/index.adoc [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+it is possible to configure the DBMS such that only one set of caches is utilized for all databases.
```
Comment (renetapopova) [inline modules/ROOT/pages/query-caches/index.adoc]:
```suggestion
```

File: modules/ROOT/pages/query-caches/index.adoc [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+More information on this topic can be found in the following section:
```
Comment (renetapopova) [inline modules/ROOT/pages/query-caches/index.adoc]:
```suggestion
For more information, see xref:query-caches/unified-query-caches.adoc[]
```

File: modules/ROOT/pages/query-caches/index.adoc [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+* xref:query-caches/unified-query-caches.adoc[]
```
Comment (renetapopova) [inline modules/ROOT/pages/query-caches/index.adoc]:
```suggestion
```

File: modules/ROOT/pages/query-caches/index.adoc [hunk 10]
```diff
@@ -1,48 +1,42 @@
...
-* xref:query-caches/unified-query-caches.adoc[]
+[[configure-caches]]
+== Configure caches
 
-[[Configuration]]
-== Configuration summary
-
-See below a summary of query cache configurations.
+The following is a summary of the query cache configurations.
+For more information, see link:{neo4j-docs-base-uri}/operations-manual/{page-version}/reference/configuration-settings/[Operations Manual -> Configuration settings].
```
Comment (sherfert) [inline modules/ROOT/pages/query-caches/index.adoc:21]: Has the operations manual been updated then?

File: modules/ROOT/pages/query-caches/index.adoc [hunk 11]
```diff
@@ -1,48 +1,42 @@
...
-| Setting | Default value | Description
+| Setting
+| Description
+| Default
 
-| `server.memory.query_cache.sharing_enabled`
-| `false`
-| label:enterprise-edition[Enterprise only] Enable sharing cache space between different databases. With this option turned on, databases will share cache space,
-but not cache entries.
+| server.memory.query_cache.sharing_enabled
```
Comment (sherfert) [inline modules/ROOT/pages/query-caches/index.adoc]: I would still find it useful if all these mentions of the exact settings were clickable links to the exact sub-section in the operations manual.

File: modules/ROOT/pages/query-caches/index.adoc (hunk 10 above)
Comment (hedengran) [reply modules/ROOT/pages/query-caches/index.adoc:21]: Yes: https://neo4j.com/docs/operations-manual/5/reference/configuration-settings/#config_server.memory.query_cache.sharing_enabled

**Review by renetapopova — APPROVED** [review]

**Review by sherfert — APPROVED** [review]
:sparkle:

**Pull request merged**